zap/src/irgen/des.rs

```diff
@@ -30,7 +36,7 @@ impl Des {
 			Enum::Unit(enumerators) => {
 				let numty = NumTy::from_f64(0.0, enumerators.len() as f64 - 1.0);
 
-				self.push_local("enum_value", Some(self.readnumty(numty)));
+				self.push_local("enum_value".into(), Some(self.readnumty(numty)));
```

Comment:
Maybe we move all vars to use `add_occurence`, or make `push_local` use it under the hood? `enum_value` can shadow as shown in [this example](https://zap.redblox.dev/playground?code=ZXZlbnQgZXhhbXBsZSA9IHsKICAgIGZyb206IFNlcnZlciwKICAgIHR5cGU6IFJlbGlhYmxlLAogICAgY2FsbDogU2luZ2xlU3luYywKICAgIGRhdGE6IGVudW0gInRlc3QiIHsKICAgICAgICBBIHsKICAgICAgICAgICAgZm9vOiBlbnVtIHsgYSwgYiwgYyB9CiAgICAgICAgfQogICAgfQp9Cg==).

Reply:
The problem with modifying `push_local` is stuff like this. `push_local` could check if the name is already suffixed and do nothing, or all variables could use `add_occurrence`.
```rs
let (len_name, len_expr) = self.add_occurrence("len");
self.push_local(
  len_name.clone(),
  Some(
	  Var::from("buffer")
		  .nindex(len_name.clone())
		  .call(vec![from_expr.clone()]),
  ),
);
```